backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/KonanCompilerFrontendServices.kt

```diff
@@ -19,7 +19,7 @@ internal fun StorageComponentContainer.initContainer(config: KonanConfig) {
 
     if (config.configuration.get(KonanConfigKeys.EMIT_LAZY_OBJC_HEADER_FILE) != null) {
         this.useImpl<ObjCExportLazyImpl>()
-        this.useInstance(ObjCExportWarningCollector.SILENT)
+        this.useInstance(ObjCExportProblemCollector.SILENT)
```

Comment:
Note that there is a semantic change here:
From now, `-Xemit-lazy-objc-header` would silently ignore exceptions. This is inline with silently ignoring warnings. Please advise, in case we should still throw exceptions here, while ignoring warnings, or in case it makes sense to report all problems similar to `ObjCExportHeaderGeneratorImpl#ProblemCollector`.

Reply:
It makes a lot of sense to make `-Xemit-lazy-objc-header` throw exceptions instead of silently ignoring them. Motivation: `-Xemit-lazy-objc-header` exists only to test `ObjCExportLazy` implementation, it is not an actual feature of the compiler. If we ignore the exceptions in tests, we will probably break something in `ObjCExportLazy` without noticing this.

Reply:
Can you please clarify whether you would like warnings to continue being ignored?

Reply:
Yes, warning should continue being ignored. So `-Xemit-lazy-objc-header` behaviour shouldn't change.